assets/js/gutenberg-plugin.js

```diff
@@ -153,24 +153,34 @@ const DistributorIcon = () => (
  * Add the Distributor panel to Gutenberg
  */
 const DistributorPlugin = () => {
-	// Ensure the user has proper permissions
-	if (
-		dtGutenberg.noPermissions &&
-		1 === parseInt( dtGutenberg.noPermissions )
-	) {
-		return null;
-	}
-
 	// eslint-disable-next-line no-shadow, react-hooks/rules-of-hooks -- permission checks are needed.
 	const postType = useSelect( ( select ) =>
 		select( 'core/editor' ).getCurrentPostType()
 	);
 
-	// eslint-disable-next-line no-shadow, react-hooks/rules-of-hooks -- permission checks are needed.
+	// eslint-disable-next-line no-shadow -- permission checks are needed.
 	const postStatus = useSelect( ( select ) =>
 		select( 'core/editor' ).getCurrentPostAttribute( 'status' )
 	);
 
+	// eslint-disable-next-line @wordpress/no-unused-vars-before-return
+	const distributorTopMenu = document.querySelector(
+		'#wp-admin-bar-distributor'
+	);
+
+	// eslint-disable-next-line no-shadow -- permission checks are needed.
+	const post = useSelect( ( select ) =>
+		select( 'core/editor' ).getCurrentPost()
+	);
+
+	// Ensure the user has proper permissions
```

Comment:
May as well fix the comment style while we're changing this code.
```suggestion
	// Ensure the user has proper permissions.
```